x-pack/plugins/ingest_manager/server/services/epm/registry/index.ts

```diff
@@ -66,9 +75,16 @@ export async function fetchFile(filePath: string): Promise<Response> {
   return getResponse(`${registryUrl}${filePath}`);
 }
 
-export async function fetchCategories(): Promise<CategorySummaryList> {
+export async function fetchCategories(params?: CategoriesParams): Promise<CategorySummaryList> {
   const registryUrl = getRegistryUrl();
-  return fetchUrl(`${registryUrl}/categories`).then(JSON.parse);
+  const url = new URL(`${registryUrl}/categories`);
+  if (params) {
+    if (params.experimental) {
+      url.searchParams.set('experimental', params.experimental.toString());
+    }
+  }
```

Comment:
We should be able to skip the individual tests by passing a record to URLSearchParams like

```suggestion
  const url = new URL(`${registryUrl}/categories`);
  if (params) {
    url.search = new URLSearchParams(params)
  }
```

If we only want some properties from `params` we can filter them

Reply:
`params.experimental` comes in as a `boolean`. `searchParams` doesn't like that, hence the need for conversion to string with `params.experimental.toString()`. I think your suggestion would run into the same issue. this could be fixed by doing `params.map()` or similar to convert boolean values, but I'll defer this improvement for now as we're only dealing with one param here atm